src/brewHandler.h

```diff
@@ -314,9 +335,17 @@ void brew() {
         case kWaitBrew: // waiting time brew
             lastBrewTime = timeBrewed;
 
+            if (FEATURE_SCALE == 1) {
+                weightBrew = weight - weightPreBrew;
+            }
+
             if (timeBrewed > totalBrewTime) {
                 currBrewState = kBrewFinished;
             }
+            // stop brew if target-weight is reached before target-time.
+            else if ((FEATURE_SCALE == 1) && (weightBrew > (weightSetpoint - scaleDelayValue))) {
```

Comment:
I think this condition should not check `FEATURE_SCALE` but `BREWCONTROL_TYPE == 2` -- otherwise, stopping the brew might be unexpected behaviour when a scale is connected but brewing is supposed to only stop by time. 

Similarly, I think that the lines above should be behind a BREWCONTROL_TYPE == 1 flag...